src/Cardano/Shell/Update/Types.hs

```diff
@@ -0,0 +1,114 @@
+module Cardano.Shell.Update.Types where
+
+import           Cardano.Prelude
+
+import qualified Data.Map as M
+
+import           Test.QuickCheck
+
+
+-- Types we need to check the behaviour.
+data Blockchain = Blockchain { getBlockchainContents :: !(Map Epoch [Slot]) }
+    deriving (Eq, Show)
+
+data Epoch = Epoch Int
+    deriving (Eq, Ord, Show)
+
+data Slot = Slot (Maybe Block)
+    deriving (Eq, Show)
+
+data Block = Block BlockHash (Maybe InstallerVersion)
+    deriving (Eq, Show)
+
+data BlockHash = BlockHash !Text
+    deriving (Eq, Show)
+
+data InstallerVersion = InstallerVersion InstallerHash
+    deriving (Eq, Ord, Show)
+
+data InstallerHash = InstallerHash !Text
+    deriving (Eq, Ord, Show)
+
+-- | The wrapper is used since this is an important distinction.
+newtype LatestInstallerVersion = LatestInstallerVersion
```

Comment:
You should wrap it with newtype instead of data.
Same goes to Epoch, Slot, BlockHash, InstallerVersion.

Reply:
Usually. Since this is not a part of any actual production code, it really isn't that critical.
It's been left this way, since it's easier to change in this form. Using a newtype wrapper removes the ability to add additional constructors...